pieforproviders/pieforproviders
# Pie 2632/two absences in a row

## 💅🏼 What issue does this fix?
Closes #2632, which is a bug that prevents the editing of two absences in a row.

## 🧳 Steps to test
1. Go to Edit Attendance
2. Click on an absence
3. Change it to a check in & check out time and hit save
4. Go to another absence
5. Try to change it to check in & check out time.
6. See that you are unable to make that change.

## 🕯 Caveats, concerns
AttendanceDataCell to edit an attendance does not automatically get its state reset once its parent EditAttendanceModal, is hidden, so things like checkInSelected and checkOutSelected remain true after setting the check in and check out times of another attendance record. I added an effect hook that forces these variables to be reset once the record being supplied to EditAttendanceModal is changed. I think there might be a way to make this a lot more performant by removing the states in AttendanceDataCell, but that might require an overhaul of the code in there.
